src/backends/tensorrt/plan_backend.cc

```diff
@@ -561,16 +549,15 @@ PlanBackend::Context::InitOptimizationProfiles(
         continue;
       }
       if (profile_index == 0) {
-        res.first->second.context_ = default_trt_context;
-        default_trt_context = nullptr;
+        res.first->second.context_ = std::move(default_trt_context);
       } else {
-        res.first->second.context_ = engine_->createExecutionContext();
+        res.first->second.context_.reset(engine_->createExecutionContext());
         if (res.first->second.context_ == nullptr) {
           return Status(
               Status::Code::INTERNAL, "unable to create TensorRT context");
         }
-        if (!res.first->second.context_->setOptimizationProfile(
-                profile_index)) {
+        if (!res.first->second.context_->setOptimizationProfileAsync(
```

Comment:
So this is the only API change needed?

Reply:
The destroy functions are also marked deprecated in TRT 8. In TRT 8 style, the runtime / engine / context are destroyed via their destructor, which is hidden in TRT 7.